feat(UserNicknamesPanel): add ability to set IRC display nickname

SuperManifolds opened the pull request:
<!--
  ⚠⚠ Do not ignore this template! ⚠⚠
  Pull requests that do not follow this format will be rejected!
-->

<!--
  Add the issues this PR closes below.
  Additional issues should be defined with a full "Closes #[issue]" line to ensure GitHub automation operates as-expected.
-->

Closes #448 

## Description
<!-- Provide a summary of changes for this pull request, and why they were implemented this way -->
- Add setDisplayNickname action to call PUT /nicknames/:id/display endpoint
- Add star icon button to set any nickname as display nickname
- Update UI to show star/delete buttons only for non-display nicknames
- Add faStar icon to fontawesome library
- Refresh user profile after setting display nickname to update UI

### Additional Information
<!-- Add any other context or development notes can go here. -->

*

### Demo
<!--
  In order to assist reviewers, Please include a screenshot or short video clip demonstrating your change in the details block below.
  If this PR does not contain any functional code change, this section may be removed.
-->
https://fuelrats.cloud/s/4NiNcdZXgfTLFsH

## Acceptance Tasks

### Contributor's Checklist
<!-- All items must be completed for the PR to be accepted -->

* [x] This PR was created to resolve an existing issue or set of issues.
* [x] This PR satisfies any and all acceptance criteria laid out by issue(s) it resolves.
* [x] I have discussed creating this PR with the maintainers in the issue(s) beforehand.
* [x] I have thoroughly tested the changes this PR introduces in a local development environment.
* [x] I have linted the entire codebase using `yarn lint` and confirmed there are no errors or warnings.
* [x] I have followed the commit conventions laid out by [`CONTRIBUTING.md`](https://github.com/FuelRats/fuelrats.com/blob/develop/CONTRIBUTING.md#commit-conventions).

### Maintainer's Checklist
<!--
  The following items must be completed by project maintainers. (A.K.A. The TechRats)
  If you are not a TechRat, do not fill out this list.
-->

* [ ] User-facing changes this PR introduces have been captured in a changeset.
* [ ] This PR has been linted and tested locally as a part of the review process.
* [ ] All issues this PR resolves have been properly linked before merging.

<!-- FIRST TIME CONTRIBUTORS -->
<!-- If you do not wish to be added to our all-contributors list, remove the "<!--" at the start of the line below -->

<!-- **I do not wish to be included as a contributor at this time** <!---->

SuperManifolds committed:
feat(UserNicknamesPanel): add ability to set IRC display nickname

- Add setDisplayNickname action to call PUT /nicknames/:id/display endpoint
- Add star icon button to set any nickname as display nickname
- Update UI to show star/delete buttons only for non-display nicknames
- Add faStar icon to fontawesome library
- Refresh user profile after setting display nickname to update UI

SuperManifolds committed:
chore: add changeset for IRC display nickname feature

SuperManifolds committed:
fix(UserNicknamesPanel): remove trailing spaces to fix eslint

UncleClapton commented on src/components/UserNicknamesPanel/UserNicknamesPanel.js:
```diff
@@ -83,24 +96,42 @@ function UserNicknamesPanel () {
-                    )
-                  }
+                  <div>
+                    {
+                      // Only show set display button for non-display nicks
+                      !isDisplayNick && (
+                        <ConfirmActionButton
+                          className="icon"
+                          confirmButtonText={`Set '${nickname.attributes?.nick}' as display nickname`}
+                          confirmSubText=""
+                          denyButtonText="Cancel"
+                          name={nickname.id}
+                          onConfirm={handleSetDisplayNickname}
+                          onConfirmText="">
+                          <FontAwesomeIcon fixedWidth icon="star" />
+                        </ConfirmActionButton>
+                      )
+                    }
+                    {
+                      // Only render for additional nicks, prevent for display nick.
+                      !isDisplayNick && (
+                        <ConfirmActionButton
+                          className="icon"
+                          confirmButtonText={`Delete nickname '${nickname.attributes?.nick}'`}
+                          confirmSubText=""
+                          denyButtonText="Cancel"
+                          name={nickname.id}
+                          onConfirm={handleDeleteNickname}
+                          onConfirmText="">
+                          <FontAwesomeIcon fixedWidth icon="trash" />
+                        </ConfirmActionButton>
+                      )
+                    }
```
since this is the same condition, I think it would be clearer to express it as a single conditional


```tsx
                    {
                      // Only show buttons for non-display nicks
                      !isDisplayNick && (
                        <>
                          <ConfirmActionButton
                            className="icon"
                            confirmButtonText={`Set '${nickname.attributes?.nick}' as display nickname`}
                            confirmSubText=""
                            denyButtonText="Cancel"
                            name={nickname.id}
                            onConfirm={handleSetDisplayNickname}
                            onConfirmText="">
                            <FontAwesomeIcon fixedWidth icon="star" />
                          </ConfirmActionButton>
                          <ConfirmActionButton
                            className="icon"
                            confirmButtonText={`Delete nickname '${nickname.attributes?.nick}'`}
                            confirmSubText=""
                            denyButtonText="Cancel"
                            name={nickname.id}
                            onConfirm={handleDeleteNickname}
                            onConfirmText="">
                            <FontAwesomeIcon fixedWidth icon="trash" />
                          </ConfirmActionButton>
                        </>
                      )
                    }
```

UncleClapton commented on src/components/UserNicknamesPanel/UserNicknamesPanel.module.scss:
```diff
@@ -21,6 +21,11 @@
     word-break: break-word;
   }
+
+  div {
+    display: flex;
+    gap: 0.5rem;
+  }
```
we should avoid adding further structure-bound styles and instead define this as it's own style at the top level of the file. maybe something like `.controlContainer`?

UncleClapton commented on .changeset/irc-display-nickname.md:
```diff
@@ -0,0 +1,9 @@
+
+- Users can now select which of their registered IRC nicknames should be their display nickname
+- Added a star icon button next to each non-display nickname to set it as the display nickname
+- The current display nickname is indicated by the absence of action buttons
```
Thinking on this more, I think this could be potentially confusing to a newcomer.

What if we used a filled in star next to the display name, and then the "set display" button uses an outline star? https://fontawesome.com/v5/icons/star?f=classic&s=regular

UncleClapton replied:
perhaps we also give that "display" star some description on hover to explain what it means

UncleClapton replied:
Could we update this to reflect the final UX?

SuperManifolds committed:
refactor(UserNicknamesPanel): address PR review feedback

- Consolidate conditional rendering for nickname buttons
- Use filled/outline star icons to show display status
- Add hover tooltips for icon buttons
- Replace generic div selector with .controlContainer class

SuperManifolds committed:
fix: resolve eslint and stylelint issues

- Fix prop ordering in UserNicknamesPanel component
- Fix color hex case and property order in styles

UncleClapton commented on src/util/fontawesome/library.js:
```diff
@@ -36,6 +36,7 @@ export {
   faSignature,
   faSpaceShuttle,
   faSpinner,
+  faStar,
```
we also need to provide the regular version, since this only provides the solid version. Sorry I forgot we don't actually have this icon pack in here yet. without adding it the icon won't load properly.

<img width="256" height="142" alt="Screenshot 2025-09-07 at 11 01 08" src="https://github.com/user-attachments/assets/aefe05ba-36fa-4e4c-8a66-3a5ac3f439f9" />


To fix:

1. run `yarn add "@fortawesome/free-regular-svg-icons@^5.15.4"`
2. add export to `library.js`

```ts
export {
  faStar as farStar,
} from '@fortawesome/free-regular-svg-icons'
```

UncleClapton commented on src/components/UserNicknamesPanel/UserNicknamesPanel.js:
```diff
@@ -57,6 +57,19 @@ function UserNicknamesPanel () {
+    if (!isError(response)) {
+      // Refresh user profile to get updated nicknames
+      await dispatch(getUserProfile())
+    }
```
Sorry I forgot about this, but because the way I handled loading states with confirmation buttons is weird, this function always needs to `return true`, or at least some truthy value. Otherwise we get an infinite spinner.

UncleClapton commented on src/components/UserNicknamesPanel/UserNicknamesPanel.js:
```diff
@@ -57,6 +57,19 @@ function UserNicknamesPanel () {
     return undefined
```
this is unrelated to your changes, but could you add `return true` here too?

UncleClapton commented on src/components/UserNicknamesPanel/UserNicknamesPanel.js:
```diff
@@ -57,6 +57,19 @@ function UserNicknamesPanel () {
+
+    const response = await dispatch(setDisplayNickname(nickname.id, nickname.attributes.nick))
+
+    if (!isError(response)) {
```
We could file this as a future enhancement instead, but I think we could easily add error handling  here. We already show errors for deletion, so we can re-use the `error` `useState` for that without changing the UI.

If we do change this now, could we also add `setError(null)` at the top of both action handlers? That way we reset the error state making it clear that if the error repeats, it's the error for the latest user interaction, and not a previous one.

UncleClapton reviewed (changes requested):
Hey Ran into a 500 from the API when setting the display nick. Would that just be because we're in lowers?

error id: `71de695d-0143-48c6-ba46-ca2608529495`

Glad it did error though, because I entirely forgot controlling the loading state for action buttons is wonky.

SuperManifolds committed:
refactor(UserNicknamesPanel): improve error handling and icon implementation

- Add @fortawesome/free-regular-svg-icons for outline star icon
- Use filled/outline stars to distinguish current vs settable display nicknames
- Add hover tooltips to icons for better UX
- Replace custom error handling with standardized getResponseError utility
- Create NicknameErrorBox for user-friendly error messages (404, 409, 422)
- Clear errors at start of operations to prevent stale error states
- Fix loading states to return proper boolean values

SuperManifolds commented:
I suspect the error 500 is due to funkiness with IRC server interaction on the dev API, I had already tested this locally and it works, I will look into why it's not working on the dev api

SuperManifolds committed:
fix(UserNicknamesPanel): use correct FontAwesome icon reference for outline star
